client/src/components/mapping/ConceptMap/index.tsx

```diff
@@ -0,0 +1,569 @@
+  id: string;
+  group: Group[];
+}
+
+interface Group {
+  source: string;
+  target: string;
+  element: Element[];
+}
+
+interface Element {
+  code: string;
+  target: Target[];
+}
+
+interface Target {
+  code: string;
+  equivalence: string;
+}
+
+interface MapRow {
+  source: string;
+  equivalence: string;
+  target: string;
+}
+
+const ConceptMap = ({ isOpen, onClose, updateInputCallback }: Props) => {
+  const [existingCodeSystems, setExistingCodeSystems] = useState(
+    [] as CodeSystem[]
+  );
+  const [existingConceptMaps, setExistingConceptMaps] = useState(
+    [] as ConceptMap[]
+  );
+  const [conceptMap, setConceptMap] = useState([] as MapRow[]);
+  const [existingConceptMapId, setExistingConceptMapId] = useState(
+    undefined as string | undefined
+  );
+  const [sourceSystemName, setSourceSystemName] = useState('');
+  const [selectedTargetSystem, setSelectedTargetSystem] = useState('');
+  const [conceptMapTitle, setConceptMapTitle] = useState('');
+  const [conceptMapName, setConceptMapName] = useState('');
+  const [conceptMapDescription, setConceptMapDescription] = useState('');
+  const [creatingNewCodeSystem, setCreatingNewCodeSystem] = useState(false);
+  const [modifyAnyway, setModifyAnyway] = useState(false);
+
+  useEffect(() => {
+    // fetch code systems
+    try {
+      axios.get(`${FHIR_API_URL}/CodeSystem`).then(response => {
+        setExistingCodeSystems(response.data.items);
+      });
+    } catch (err) {
+      console.error(
+        `Could not fecth code systems: ${
+          err.response ? err.response.data : err.message
+        }`
+      );
+    }
+    // fetch concept maps
+    try {
+      axios.get(`${FHIR_API_URL}/ConceptMap`).then(response => {
+        setExistingConceptMaps(response.data.items);
+      });
+    } catch (err) {
+      console.error(
+        `Could not fecth concept maps: ${
+          err.response ? err.response.data : err.message
+        }`
+      );
+    }
+  }, []);
+
+  // TODO these are actually codes from a code system. Fetch them instead of hard coding them
+  const choiceEquivalences = [
+    'relatedto',
+    'equivalent',
+    'equal',
+    'wider',
+    'subsumes',
+    'narrower',
+    'specializes',
+    'inexact',
+    'unmatched',
+    'disjoint'
+  ];
+
+  const displayCode = (code: string): string => (code ? code : 'Select code');
+
+  const isSourceSystemDisabled = (system: CodeSystem): boolean =>
+    system.title === selectedTargetSystem;
+
+  const isTargetSystemDisabled = (system: CodeSystem): boolean =>
+    system.title === sourceSystemName;
+
+  const emptyFields = conceptMap.some(
+    row => !row.source || !row.equivalence || !row.target
+  );
+
+  const resetMap = (): void => {
+    setConceptMap([]);
+    setConceptMapName('');
+    setConceptMapTitle('');
+    setConceptMapDescription('');
+    setModifyAnyway(false);
+    setExistingConceptMapId(undefined);
+  };
+
+  const fetchCodesForCodeSystem = (systemName: string): string[] =>
```

Comment:
fetch make me think of an HTTP call, how do you feel about renaming this to something like `(find|get|extract)CodesFromSystem`